kernel/sched.c

```diff
@@ -975,7 +975,15 @@ s32_t z_impl_k_sleep(int ms)
 {
 	s32_t ticks;
 
-	ticks = k_ms_to_ticks_ceil32(ms);
+	__ASSERT((ms >= 0) || (ms == K_FOREVER),
+		 "Only non-negative values are accepted.");
+	/* If asserts are disabled treat negative numbers as the past. */
+	if ((ms < 0) && (ms != K_FOREVER)) {
```

Comment:
This behavior needs to be documented, basically now the API is accepting non-negative values and treating them. This is where we should have asserts and error checking consistent, we can't have both in such an inconsistent way. Either we do not accept non-negative, or we do...

Reply:
We do not accept but we used to. Before #19591 negative values would result in immediate exit. It is added to maintain the same behavior for those who used it that way. I personally, have no strong opinion about it and can remove it. It was added per @pabigot request.

Reply:
Well, can we at least document this fact in the doxygen @nordic-krch? The fact that negative values mean `k_sleep()` exits immediately.

Reply:
I think that would be acceptable.

Reply:
added doxygen information.

Reply:
> This behavior needs to be documented, basically now the API is accepting non-negative values and treating them. This is where we should have asserts and error checking consistent, we can't have both in such an inconsistent way. Either we do not accept non-negative, or we do...

This is now documented @nashif

Reply:
ok, thanks for the doc, but now that it is documented as "Negative value will result in immediate return." why do we assert on negative values? This was my second point in the original comment.